### PR TITLE
Rework Symfony dependencies + SF4 compat

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targeting this branch, because this is BC.

## Changelog

```markdown
### Added
- Compatibility with Symfony4
```

## Subject

Remove direct dependency with framework bundle (only needed for test), and make Symfony dependencies explicit.

Also allow symfony4 to be installed

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -31,7 +35,8 @@
         "matthiasnoback/symfony-dependency-injection-test": "^1.0",
         "sonata-project/doctrine-orm-admin-bundle": "^3.2",
         "stof/doctrine-extensions-bundle": "^1.1",
-        "symfony/phpunit-bridge": "^3.3"
+        "symfony/framework-bundle": "^2.8 || ^3.2 || ^4.0",
+        "symfony/phpunit-bridge": "^3.3 || ^4.0"
```
Comment (core23) [inline composer.json]: Can we drop 3.3 here?

Comment (jordisala1991) [reply composer.json]: Would be better to do it when we have a reason. when we add the listener on the phpunit.xml.dist for example

Comment (OskarStark) [reply composer.json]: this will be done by dev-kit in the near future

Comment (jordisala1991) [reply composer.json]: Then when dev-kit creates the PR we modify it changing this version

**Pull request merged**